src/Validation/RulesProvider.php

```diff
@@ -59,9 +59,9 @@ public function __construct($class = Validation::class)
      *
      * @param string $method the validation method to call
      * @param array $arguments the list of arguments to pass to the method
-     * @return bool whether or not the validation rule passed
+     * @return mixed whether or not the validation rule passed
```

Comment:
You should check/revert all mixed changes.

Reply:
All mixed changes are based on actual code. Adding typehints as per old docblocks causes tests to fail.

Reply:
What does a method return that so far was boolean? See comment beside it.

Reply:
This is for the magic `__call()` which proxies calls to other to other classes so we really can't guarantee that the return type will always be boolean.

Reply:
No one can ever guarantee that, nor did we ever do that.
We always have been annotating the expected input and output types, that is what we should continue to do, not all possible wrong ones on top. Otherwise documentation here really doesnt help much, neither for humans nor for tooling.

Reply:
To clarify: The follow up code will blow up if the documented expectations are not met. That is then outside of this documentation here. 
We could, if we want, also add a bool typehint to assert that on code level of course.

Reply:
Same comment as above. I fixed the incorrect type, you can update it for accuracy.